### PR TITLE
GameProfileManager documentation

### LINKED ISSUE #458: Documentation on GameProfileCache
Document the [GameProfileCache](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/profile/GameProfileCache.java), including what it is, how to use it, etc...

Relevant Commit(s): https://github.com/SpongePowered/SpongeAPI/commit/8a1f85c921fb6b1981e414b6ddbb49d290202cde
Relevant PR: https://github.com/SpongePowered/SpongeAPI/pull/1068

### PR DESCRIPTION
**Pull request by RobertHerhold**
Resolves #458, #461.

### CODE REVIEW

File: source/plugin/game-profile-manager.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Creating GameProfiles
+=====================
+
+You can generate a new ``GameProfile`` using ``GameProfileManager#createProfile(UUID, String)``. Note that the username
+does not necessarily need to correspond to the ``UUID`` of that player. Likewise, the ``UUID`` does not need to belong
+to a valid player.
+
+.. code-block:: java
+
+    GameProfile gameProfile =
```
Comment (kashike) [inline source/plugin/game-profile-manager.rst]: Recommend using `GameProfile.of(UUID, String)` instead.

File: source/plugin/game-profile-manager.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+You can also directly access the ``GameProfileCache`` used by Sponge to store ``GameProfile``\ s. To do so, simply call
+the ``GameProfileManager#getCache`` method. Using the ``GameProfileCache``, you can look up ``GameProfile``\ s, add newly
+constructed ``GameProfile``\ s, and fill profiles with data stored in the cache.
+
+.. code-block:: java
+
+    import org.spongepowered.api.profile.GameProfileCache;
+
+    GameProfile fakeProfile =
```
Comment (kashike) [inline source/plugin/game-profile-manager.rst]: Recommend using `GameProfile.of(UUID, String)` instead.

File: source/plugin/game-profile-manager.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    import java.util.Collection;
+
+    profile.getPropertyMap().put(
+        "key", profileManager.createProfileProperty("foo", "bar", null));
+    Collection<ProfileProperty> customProperties = profile.getPropertyMap().get("key");
+
+Working with the GameProfileCache
+=================================
+
```
Comment (kashike) [inline source/plugin/game-profile-manager.rst]: It should be documented that `GameProfileCache` can be replaced by plugins, as well.

File: source/plugin/game-profile-manager.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    GameProfile filledProfile = profileManager.fill(gameProfile).get();
+
+Working with ProfileProperties
+==============================
+
+``GameProfile``\ s can be used to store arbitrary data about a player using ``ProfileProperty``\ s. However, this cannot
+not be used as a permanent data store, as the data is not persisted across server restarts. We can retrieve the properties
+of a ``GameProfile`` using the ``GameProfile#getPropertyMap`` method, which returns a ``Multimap``. From there, you can
+retrieve existing or store new ``ProfileProperty``\ s, which are represented as a key value pair. To generate a new
+``ProfileProperty``, simply call the ``GameProfileManager#createProfileProperty(String, String, @Nullable String)``
```
Comment (kashike) [inline source/plugin/game-profile-manager.rst]: `ProfileProperty.of(...)`

File: source/plugin/game-profile-manager.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+    import java.util.concurrent.CompletableFuture;
+
+    CompletableFuture<GameProfile> futureGameProfile = profileManager.get("Notch");
+
+Retrieving by UUID
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: java
+
```
Comment (12AwsomeMan34) [inline source/plugin/game-profile-manager.rst]: Would be nice to have an import for UUID.

File: source/plugin/game-profile-manager.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    GameProfile filledProfile = profileManager.fill(gameProfile).get();
+
+Working with ProfileProperties
+==============================
+
+``GameProfile``\ s can be used to store arbitrary data about a player using ``ProfileProperty``\ s. However, this cannot
+not be used as a permanent data store, as the data is not persisted across server restarts. We can retrieve the properties
```
Comment (Inscrutable) [inline source/plugin/game-profile-manager.rst]: The grammar is a bit weird here. "as the data does not persist between server restarts" sounds better to me.

**Pull request merged**